src/actions/search.js

```diff
@@ -29,22 +29,61 @@ const searchDelegate = ({ publicKey, address }) =>
     });
   };
 
-const searchVotes = ({ address }) =>
-  (dispatch, getState) => {
+
+export const searchVotesDelegate = (votes, {
+  showingVotes = 30, address, offset = 0, limit = 101, filter = '',
+}) =>
+  // eslint-disable-next-line max-statements
+  async (dispatch, getState) => {
     const liskAPIClient = getState().peers.liskAPIClient;
-    /* istanbul ignore else */
-    if (liskAPIClient) {
-      getAllVotes(liskAPIClient, address).then(response =>
-        dispatch({
-          type: actionTypes.searchVotes,
-          data: {
-            votes: response.data.votes,
-            address,
-          },
-        }));
+    if (!liskAPIClient) return;
+    dispatch(loadingStarted(actionTypes.searchVotes));
+    const delegates = await listDelegates(liskAPIClient, { limit, offset });
+    const votesWithDelegateInfo = votes.map((vote) => {
+      const delegate = delegates.data.find(d => d.username === vote.username) || {};
+      return { ...vote, ...delegate };
+    }).sort((a, b) => {
+      if (!a.rank && !b.rank) return 0;
+      if (!a.rank || +a.rank > +b.rank) return 1;
+      return -1;
+    });
+
+    const filteredVotes = votesWithDelegateInfo.filter(vote => RegExp(filter, 'i').test(vote.username));
+    const lastIndex = showingVotes > filteredVotes.length ?
+      filteredVotes.length : showingVotes;
+    if (filteredVotes.length && !filteredVotes[lastIndex - 1].rank) {
+      dispatch(searchVotesDelegate(votesWithDelegateInfo, {
+        offset: offset + limit,
+        address,
+        showingVotes,
+        filter,
+      }));
+    } else {
+      dispatch({
+        type: actionTypes.searchVotes,
+        data: { votes: votesWithDelegateInfo, address },
+      });
+      dispatch(loadingFinished(actionTypes.searchVotes));
     }
   };
 
+const searchVotes = ({ address, offset, limit }) =>
+  async (dispatch, getState) => {
+    const liskAPIClient = getState().peers.liskAPIClient;
+    /* istanbul ignore else */
+    if (!liskAPIClient) return;
+    dispatch(loadingStarted(actionTypes.searchVotes));
+    const votes = await getVotes(liskAPIClient, { address, offset, limit })
+      .then(res => res.data.votes);
+
+    dispatch({
+      type: actionTypes.searchVotes,
+      data: { votes, address },
```

Comment:
could you change this dispatch to instead of trigger this function in this way dispatch it in this other way

example
```
dispatch(searchVotesDelegate(votes, { address, limit: 30 }))
```

Reply:
Actually removed this one, since it was causing trouble on the old version